src/Utils/CliMsgFormatter.php

```diff
@@ -95,7 +95,26 @@ public function yellow( string $text ): string {
 	 * @return string
 	 */
 	public function progress( int $i, int $total ): string {
-		return min( 100, round( ( $i / $total ) * 100 ) ) . " %";
+		// @codeCoverageIgnoreStart
+		return $this->calculateProgress( $i, $total, true );
+		// @codeCoverageIgnoreEnd
+	}
+
+	/**
+	 * @codeCoverageIgnore
+	 */
+	private function calculateProgress( int $i, int $total, bool $percentage ) {
+		$value = 100;
+
+		if ( $i > 0 && $total > 0 ) {
+			$value = min( 100, round( ( $i / $total ) * 100 ) );
+		}
+
+		if ( $percentage ) {
+			$value .= " %";
```

Comment:
Did you see this part? I don't think the way you did the extraction makes sense and is correct.

Better if `calculateProgress` returns `int` (it can have this PHP return type) and that `progress` appends the ` %`. `progressCompact` expects an int.

Reply:
Thanks, done.